File: src/main/java/br/com/kronos/backend/aplicacao/pessoa/Titulacao.java
package br.com.kronos.backend.aplicacao.pessoa;

import static br.com.kronos.backend.aplicacao.validador.Validacoes.exigirNaoNulo;
import static br.com.kronos.backend.aplicacao.validador.Validacoes.exigirTamanho;

import java.time.LocalDate;

import lombok.Builder;
import lombok.Getter;

@Getter
public class Titulacao {
	
	private int id;
    private LocalDate data;
    private String nomeTitulo;
    private Integer idArqAnexo;
    private Long idPessoa;
    private Integer idEmpresa;
    private Integer idTipoTitulo; 

	@Builder
	public Titulacao(int id, LocalDate data, String nomeTitulo, Integer idArqAnexo, Long idPessoa, Integer idEmpresa, Integer idTipoTitulo) {

        this.id = id;
        this.data = exigirNaoNulo(data, "data é obrigatório");
        this.nomeTitulo = exigirTamanho(nomeTitulo, "Nome do Título", 1, 100);
        this.idArqAnexo = idArqAnexo;
        this.idPessoa = exigirNaoNulo(idPessoa, "Pessoa"); 
        this.idEmpresa = exigirNaoNulo(idEmpresa, "Empresa");
        this.idTipoTitulo = exigirNaoNulo(idTipoTitulo, "Tipo do Título");
       
	}
}

File: poc-kronos-frontend/src/model/pessoa/Titulacao.js
class Titulacao {
  constructor(dados) {
    this.id = dados ? dados.id : null;
    this.data = dados ? dados.data : null;
    this.nomeTitulo = dados ? dados.nomeTitulo : null;
    this.idArqAnexo = dados ? dados.idArqAnexo : null;
    this.idPessoa = dados ? dados.idPessoa : null;
    this.idEmpresa = dados ? dados.idEmpresa : null;
    this.nomeEmpresa = dados ? dados.nomeEmpresa : null;
    this.idTipoTitulo = dados ? dados.idTipoTitulo : null;
    this.nomeTipoTitulo = dados ? dados.nomeTipoTitulo : null;
  }

  static validations(required, maxLength) {
    return {
      data: { required },
      nomeTitulo: { required, maxLength: maxLength(100) },
      idPessoa: { required },
      idEmpresa: { required },
      idTipoTitulo: { required }
    };
  }

  static validationMessages() {
    return {
      data: {
        required: "Data é obrigatória"
      },
      nomeTitulo: {
        required: "Nome Título é obrigatório",
        maxLength: "Nome Título só aceita no máximo 100 caracteres"
      },
      idPessoa: {
        required: "Pessoa é obrigatória"
      },
      idEmpresa: {
        required: "Empresa é obrigatório"
      },
      idTipoTitulo: {
        required: "Tipo de Título é obrigatório"
      }
    };
  }
}
export default Titulacao;
